Client/Assets/NightModeHelper.js

```diff
@@ -6,39 +6,94 @@
   if (!window.__firefox__) {
     window.__firefox__ = {};
   }
+  var observers = [];
   var isEnabled = false;
 
   function initializeStyleSheet (enabled) {
-    var night_css_id = "iBrowser_night_css";
-    var night_css = ':not(body){color: #C8C8C8 !important; background-color:transparent !important;}body{-webkit-filter:brightness(.5) !important; background-color:#202020 !important;}';
-    var newCss = document.getElementById(night_css_id);
+    var night_mode_css_id = "iBrowser_night_mode_css";
+    var night_mode_css = '*{color: #C8C8C8 !important;}body,.nightMode{background-color:#202020 !important;}body{-webkit-filter:brightness(.5) !important;}';
+    var newCss = document.getElementById(night_mode_css_id);
     if (enabled) {
-        if (!newCss) {
-            var cssStyle = document.createElement('style');
-            cssStyle.type = 'text/css';
-            cssStyle.id = night_css_id;
-                if (cssStyle.styleSheet) {
-                    cssStyle.styleSheet.cssText = night_css;
-                } else {
-                    cssStyle.appendChild(document.createTextNode(night_css));
-                }
-            document.documentElement.appendChild(cssStyle);
+      if(!newCss){
+        var cssStyle = document.createElement('style');
+        cssStyle.type = 'text/css';
+        cssStyle.id = night_mode_css_id;
+        if (cssStyle.styleSheet) {
+          cssStyle.styleSheet.cssText = night_mode_css;
         } else {
-            newCss.innerHTML = night_css;
+          cssStyle.appendChild(document.createTextNode(night_mode_css));
         }
+        document.documentElement.appendChild(cssStyle);
+      } else {
+        newCss.innerHTML = night_mode_css;
+      }
     } else {
-        if (newCss) {
-            newCss.innerHTML = "";
-        }
+      if (newCss) {
+        newCss.innerHTML = "";
+      }
+    }
+    var bodys = document.getElementsByTagName("BODY");
+    for (var i=0; i<bodys.length; i++) {
+        blockBackground(bodys[i], enabled);
+        watchForElems(bodys[i], enabled);
+    }
+  }
+
+  function blockBackground (elem, enabled) {
```

Comment:
I am very concerned that this function is doing the job of cascading styles, but in Javascript instead of CSS.

Is there a way of making the CSS work harder, and the JS less hard?